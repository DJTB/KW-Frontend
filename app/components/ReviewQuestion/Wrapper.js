import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  text-align: center;
`;

export default Wrapper;
