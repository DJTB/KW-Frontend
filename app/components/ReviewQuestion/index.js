import React, { PropTypes } from 'react';
import Immutable from 'immutable';
import TagList from 'components/TagList';
import Chip from 'components/Chip';
import {
  Wrapper,
  MeaningWrapper,
  Meaning,
} from './UI';

const Question = ({ meaning, tags }) => (
  <Wrapper>
    <MeaningWrapper>
      <Meaning>{meaning}</Meaning>
    </MeaningWrapper>
    {tags && <TagList items={tags} component={Chip} />}
  </Wrapper>
  );

Question.propTypes = {
  meaning: PropTypes.string.isRequired,
  tags: PropTypes.instanceOf(Immutable.Iterable),
};

export default Question;
