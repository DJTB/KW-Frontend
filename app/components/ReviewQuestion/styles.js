import styled from 'styled-components';

import { white, purpleDark } from 'shared/styles/colors';
import { centerByMargin } from 'shared/styles/layout';
import { giga } from 'shared/styles/typography';

import TagsList from 'components/TagsList';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  text-align: center;
  flex: 1 1 100%;
`;

export const QuestionWrapper = styled.div`
  flex: 999 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const Question = styled.h1`
  ${giga}
  ${centerByMargin};
  color: ${white};
  font-weight: 700;
  letter-spacing: -0.03em;
  text-shadow: .1em .2em .4em ${purpleDark};
  padding: 2em .5em;
`;

export const Tags = styled(TagsList)`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-self: center;
  z-index: 2;
`;
