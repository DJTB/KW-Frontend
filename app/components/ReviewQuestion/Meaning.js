import styled from 'styled-components';
import { white, purpleDark } from 'shared/styles/colors';
import { media } from 'shared/styles/media';

const Meaning = styled.h1`
  color: ${white};
  text-shadow: 1px 2px 4px ${purpleDark};
  padding: 50px 15px 30px;
  display: table-cell;
  font-size: calc(20px + 30 * ((100vw - 300px) / 1700));
  vertical-align: middle;
  line-height: 1.3;
  letter-spacing: -1px;
  margin-top: .5em;
  ${media('min').lg`
    padding-left: 10%;
    padding-right: 10%;
  `}
`;

export default Meaning;
