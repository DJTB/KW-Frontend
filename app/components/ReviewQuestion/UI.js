import styled from 'styled-components';
import TagList from 'components/TagList';
import { white, purpleDark } from 'shared/styles/colors';
import { giga } from 'shared/styles/typography';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  text-align: center;
  flex: 1 1 100%;
`;

export const MeaningWrapper = styled.div`
  flex: 999 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const Meaning = styled.h1`
  ${giga}
  color: rgb(${white});
  margin: 0;
  font-weight: 700;
  letter-spacing: -0.03em;
  text-shadow: .1em .2em .4em rgb(${purpleDark});
  padding: 2em .5em;
`;

export const Tags = styled(TagList)`
  display: flex;
  align-self: flex-start;
  flex-flow: row wrap;
  justify-content: flex-start;
  z-index: 2;
`;
