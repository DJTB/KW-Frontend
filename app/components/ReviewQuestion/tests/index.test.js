import React from 'react';
import { shallow } from 'enzyme';
import ReviewQuestion from '../index';

describe('<ReviewQuestion />', () => {
  it('should match baseline snapshot', () => {
    expect(shallow(<ReviewQuestion />)).toMatchSnapshot();
  });
});
