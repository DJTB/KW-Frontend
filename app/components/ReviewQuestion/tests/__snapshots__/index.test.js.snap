exports[`<ReviewQuestion /> should render error message instead of meaning when error present 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "bsEham",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "knAAdX",
            },
            "children": Array [
              Object {
                "data": "Something went wrong: \"undefined\". Please contact us or try again!",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;

exports[`<ReviewQuestion /> should render meaning text when passed required props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "bsEham",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "knAAdX",
            },
            "children": Array [
              Object {
                "data": "test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
